### ISSUE TITLE
Scheduled Jobs are not being executed

### ISSUE THREAD

**Issue by janaurka** [starkandwayne/shield#389]
Hey,

I’m running:
* Shield 8.0.6
* Deployed through the `bosh-release`
* on vSphere 5.5/6.0

The Jobs have been configured through the WebUI, here is an example:

This is from:

`v2/tenants/<tenant-uuid>/jobs`

```
{
"uuid": "ca8227ca-19d6-471c-a687-d2631d2bc39e",
"name": "<name>",
"summary": "",
"expiry": 2592000,
"schedule": "every 6 hours from 2:15",
"paused": false,
"fixed_key": false,
"Target": {
"uuid": "2297c378-e350-459c-b467-8502c93731da",
"name": "<name>",
"agent": "",
"plugin": "mysql",
"endpoint": "{\"mysql_bindir\":\"/var/vcap/packages/mariadb/bin\",\"mysql_host\":\"<REDACTED>\",\"mysql_options\":\"--quick --single-transaction\",\"mysql_password\":\"<REDACTED>\",\"mysql_port\":\"3306\",\"mysql_user\":\"root\"}"
},
"store": {
"uuid": "3115cc1d-0124-42e0-9c27-837f6c31839b",
"name": "default-s3",
"agent": "",
"plugin": "s3",
"summary": "Default Solera S3 Storage",
"endpoint": "{\"access_key_id\":\"<REDACTED>\",\"bucket\":\"<REDACTED>\",\"s3_host\":\"<REDACTED>\",\"secret_access_key\":\"<REDACTED>\"}"
},
"policy": {
"uuid": "6d277c67-85fd-4417-9c6a-76302904d878",
"name": "30d",
"summary": "30 Days"
},
"agent": "<REDACTED>:5444",
"last_run": 1516012044,
"last_task_status": "done"
},
```
-> `last_run` -> `Mon, 15 Jan 2018 10:27:24 GMT` which was a `force-run` triggered through the WebUI by myself.

If I check the tasks which have been executed, there is nothing listed:

```
587341ab-78ad-4769-9db7-51120941f466  test-store  done    system       2018-01-17 10:20:19+0100  2018-01-17 10:20:19+0100
0714cccd-4b0a-4a69-9a00-4bce1dfea258  test-store  done    system       2018-01-17 09:20:19+0100  2018-01-17 09:20:20+0100
047a79e2-25b7-4122-9860-2bcca4b5227f  test-store  done    system       2018-01-17 08:20:19+0100  2018-01-17 08:20:20+0100
613fe3a9-f8a5-48cf-b570-2adb8d5373e3  test-store  done    system       2018-01-17 07:20:19+0100  2018-01-17 07:20:20+0100
414a5498-3b48-43f7-9ae8-621332e97202  test-store  done    system       2018-01-17 06:20:19+0100  2018-01-17 06:20:20+0100
ba429ac0-dace-4dd2-ab26-a25dc404edcd  test-store  done    system       2018-01-17 05:20:19+0100  2018-01-17 05:20:21+0100
4358d312-f896-4144-96e2-1b34b516e37a  test-store  done    system       2018-01-17 04:20:19+0100  2018-01-17 04:20:19+0100
ef227853-8010-49bc-bd49-bfca2f8ee822  test-store  done    system       2018-01-17 03:20:19+0100  2018-01-17 03:20:19+0100
94285d5c-0f55-485f-8bb7-b87645970d1e  test-store  done    system       2018-01-17 02:20:19+0100  2018-01-17 02:20:19+0100
1250b596-21b3-4a68-acc0-817f7a7dcd65  test-store  done    system       2018-01-17 01:20:19+0100  2018-01-17 01:20:20+0100
f47ef5f4-e75a-4e07-a4cb-31529b76282b  test-store  done    system       2018-01-17 00:20:19+0100  2018-01-17 00:20:20+0100
```

although there are `>5` jobs which should trigger every 6h and some daily ones.

expected result: Backups are being executed according to the schedule.

**Comment by jhunt** [comment]
Gut feel: this is probably being caused by an oversight / mismatch in the SHIELD scheduler and the locking mechanism of the backend Vault:

1. SHIELD won't schedule tasks to run if it is locked
2. SHIELD only re-schedules tasks after they finish running (to avoid lapping)
3. When it starts up, SHIELD is always locked (by design)
4. When it starts up, SHIELD tries to schedule tasks.
5. Since the tasks never actually get schedules, per (1), they never run which means they never get re-scheduled, per (2).

(Just spitballing here; I hope to get into the code this afternoon and verify the conjecture)

**Comment by jhunt** [comment]
As usual, the actual bug was far stupider.

I believe the fix is in 302f10e

**Comment by janaurka** [comment]
Update to `8.0.8` applied on Saturday and since then the backups are scheduled and executed as expected. Great work, thx!
